pkg/scheduler/queue/tree_queue_algo_querier_worker_queue_priority_test.go

```diff
@@ -0,0 +1,345 @@
+	querierWorkerPrioritizationQueueAlgo := NewQuerierWorkerQueuePriorityAlgo()
+
+	tree, err := NewTree(&roundRobinState{}, querierWorkerPrioritizationQueueAlgo, &roundRobinState{})
+	require.NoError(t, err)
+
+	// enqueue 2 objects each to 2 different children, each with 3 different grandchildren;
+	// the highest layer is managed by a vanilla round-robin rotation
+	// and the second-highest layer is managed by the querier-worker queue priority algorithm.
+	// The global node order rotation for the querier-worker queue priority algorithm is only affected
+	// when the first node of a type is created in the tree layer or the last node of a type is emptied and removed.
+	// To keep things brief "i", "sg", and "isg" are used for "ingester", "store-gateway", and "ingester-and-store-gateway".
+
+	// first creation of an ingester node anywhere adds ingester node to global node order
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"a", ingesterQueueDimension}, "obj-a-i-1"))
+	// same for store-gateway node
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"b", storeGatewayQueueDimension}, "obj-b-sg-1"))
+	// same for ingester-and-store-gateway node
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"b", ingesterAndStoreGatewayQueueDimension}, "obj-b-isg-1"))
+	// further node creation of those same types does not affect the global order
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"b", ingesterQueueDimension}, "obj-b-i-1"))
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"a", storeGatewayQueueDimension}, "obj-a-sg-1"))
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"a", ingesterAndStoreGatewayQueueDimension}, "obj-a-isg-1"))
+
+	// fill in a second object for each of the 6 paths created above
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"a", ingesterQueueDimension}, "obj-a-i-2"))
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"a", storeGatewayQueueDimension}, "obj-a-sg-2"))
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"a", ingesterAndStoreGatewayQueueDimension}, "obj-a-isg-2"))
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"b", ingesterQueueDimension}, "obj-b-i-2"))
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"b", storeGatewayQueueDimension}, "obj-b-sg-2"))
+	require.NoError(t, tree.EnqueueBackByPath(QueuePath{"b", ingesterAndStoreGatewayQueueDimension}, "obj-b-isg-2"))
+
+	/* balanced tree structure before dequeuing:
+		root
+		├── a
+		│   ├── ingester
+		│	│	├── obj-a-i-1
+	    │   │   ├── obj-a-i-2
+		│   ├── store-gateway
+		│	│	├── obj-a-sg-1
+	    │   │   ├── obj-a-sg-2
+		│   └── ingester-and-store-gateway
+		│	 	├── obj-a-isg-1
+	    │       ├── obj-a-isg-2
+		├── b
+		│   ├── ingester
+		│	│	├── obj-b-i-1
+	    │   │   ├── obj-b-i-2
+		│   ├── store-gateway
+		│	│	├── obj-b-sg-1
+	    │   │   ├── obj-b-sg-2
+		│   └── ingester-and-store-gateway
+		│	 	├── obj-b-isg-1
+	    │       ├── obj-b-isg-2
```

Comment:
Not sure what Github is doing here; formatting looks weird in browser, but looks fine when I pull down the branch.

Reply:
Might be your UI setting for tab width appearance - which is at https://github.com/settings/appearance

Reply:
You're right; mine was defaulted to 8